plugins/sentinelone/komand_sentinelone/tasks/monitor_activities_and_events/task.py

```diff
@@ -0,0 +1,327 @@
+    EVENTS_LOGS: {"LAST_LOG_TIMESTAMP": EVENTS_LAST_LOG_TIMESTAMP, "PAGE_TOKEN": EVENTS_PAGE_CURSOR},
+    THREATS_LOGS: {"LAST_LOG_TIMESTAMP": THREATS_LAST_LOG_TIMESTAMP, "PAGE_TOKEN": THREATS_PAGE_CURSOR},
+}
+
+
+class MonitorActivitiesAndEvents(insightconnect_plugin_runtime.Task):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="monitor_activities_and_events",
+            description=Component.DESCRIPTION,
+            input=MonitorActivitiesAndEventsInput(),
+            output=MonitorActivitiesAndEventsOutput(),
+            state=MonitorActivitiesAndEventsState(),
+        )
+
+    def run(self, params={}, state={}, custom_config={}):
+        """
+        Query activities, device control events, and threats logs between a supplied timeframe and the current time.
+        Return all found logs and set last task run time, pagination cursors, and last log timestamps for each log type.
+        """
+        existing_state = state.copy()
+        try:
+            is_initial_run = self.check_initial_run(state)
+            queries = self.check_queries(params)
+            cursors = self.get_cursors(state)
+            total_queries = self.count_queries(queries)
+            total_forbidden_responses = 0
+            last_run_timestamp = state.get(LAST_RUN_TIMESTAMP)
+            self.logger.info(cursors)
+            is_paginating = any(cursors.values())
+            if is_paginating:
+                self.log_pagination_cycle(cursors)
+
+            current_run_timestamp, lookback_timestamp = self.get_lookback_values(
+                is_initial_run, custom_config, last_run_timestamp
+            )
+
+            all_logs, total_forbidden_responses = self.collect_logs_handler(
+                queries,
+                cursors,
+                state,
+                lookback_timestamp,
+                current_run_timestamp,
+                total_forbidden_responses,
+                is_paginating,
+            )
+
+            # Check if all ran queries have returned 401 or 403 errors and raise an exception if so
+            if total_forbidden_responses >= total_queries > 0:
+                return [], existing_state, False, 401, PluginException(preset=PluginException.Preset.UNAUTHORIZED)
+            has_more_pages = self.determine_next_pagination_cycle(state, lookback_timestamp, current_run_timestamp)
+            return all_logs, state, has_more_pages, 200, None
+        except ApiException as error:
+            return [], existing_state, False, error.status_code, error
+        except Exception as error:
+            return [], existing_state, False, 500, PluginException(preset=PluginException.Preset.UNKNOWN, data=error)
+
+    def check_initial_run(self, state: Dict) -> bool:
+        """
+        Check if state exists, return False if so indicating initial run
+        """
+        is_initial_run = not state
```

Comment:
Stupid question but can I check what this is doing? Is this setting is_initial_run then we are checking it on the next line?